drivers/reset/reset_mmio.c

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2025 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/reset.h>
+
+struct reset_mmio_dev_config {
+	uint32_t base;
+	uint8_t num_resets;
+	bool active_low;
+};
+
+static inline int reset_mmio_status(const struct device *dev, uint32_t id, uint8_t *status)
+{
+	const struct reset_mmio_dev_config *config = dev->config;
+	uint32_t value;
+
+	if (id >= config->num_resets) {
+		return -EINVAL;
+	}
+
+	value = sys_read32(config->base);
+	*status = FIELD_GET(BIT(id), value);
+
+	/* If active low, invert the logic */
+	if (config->active_low) {
+		*status = !*status;
+	}
+
+	return 0;
+}
+
+static inline int reset_mmio_update(const struct device *dev, uint32_t id, uint8_t assert)
+{
+	const struct reset_mmio_dev_config *config = dev->config;
+	uint32_t value;
+	bool set;
+
+	if (id >= config->num_resets) {
+		return -EINVAL;
+	}
+
+	/* If active low, invert the logic */
+	set = config->active_low ? !assert : assert;
+
+	value = sys_read32(config->base);
+
+	if (set) {
+		value |= BIT(id);
+	} else {
+		value &= ~BIT(id);
+	}
+
+	sys_write32(value, config->base);
+
+	return 0;
+}
+
+static int reset_mmio_line_assert(const struct device *dev, uint32_t id)
+{
+	return reset_mmio_update(dev, id, 1);
+}
+
+static int reset_mmio_line_deassert(const struct device *dev, uint32_t id)
+{
+	return reset_mmio_update(dev, id, 0);
+}
+
+static int reset_mmio_line_toggle(const struct device *dev, uint32_t id)
+{
+	uint8_t reset_status = 0;
+	int status;
+
+	status = reset_mmio_status(dev, id, &reset_status);
+	if (status) {
+		return status;
+	}
+
+	return reset_mmio_update(dev, id, !reset_status);
+}
+
+static const struct reset_driver_api reset_mmio_driver_api = {
```

Comment:
```suggestion
static DEVICE_API（reset，reset_mmio_driver_api） = {
```

Use `DEVICE_API` macro.

Reply:
Done